flake.nix

```
@@ -4,19 +4,36 @@
 
   outputs = { self, nixpkgs }: {
     lib = {
-      # TODO handle multiple architectures.
-      evalConfig = import ./eval-config.nix { inherit (nixpkgs) lib; };
-
-      darwinSystem =
-        { modules, inputs ? { }
-        , system ? throw "darwin.lib.darwinSystem now requires 'system' to be passed explicitly"
-        , ...
-        }@args:
-        self.lib.evalConfig (args // {
-          inherit system;
-          inputs = { inherit nixpkgs; darwin = self; } // inputs;
-          modules = modules ++ [ self.darwinModules.flakeOverrides ];
-        });
+      evalConfig = import ./eval-config.nix;
+
+      darwinSystem = args@{ modules, ... }: self.lib.evalConfig (
+        { inherit (nixpkgs) lib; }
+        // nixpkgs.lib.optionalAttrs (args ? pkgs) { inherit (args.pkgs) lib; }
+        // builtins.removeAttrs args ["system" "pkgs" "inputs"]
+        // {
+          modules = modules
+            ++ nixpkgs.lib.optional (args ? pkgs) ({ lib, ... }: {
+              _module.args.pkgs = lib.mkForce args.pkgs;
+            })
+            # Backwards compatibility shim; TODO: warn?
+            ++ nixpkgs.lib.optional (args ? system) ({ lib, ... }: {
+              nixpkgs.system = lib.mkDefault args.system;
+            })
+            # Backwards compatibility shim; TODO: warn?
+            ++ nixpkgs.lib.optional (args ? inputs) {
+              _module.args.inputs = args.inputs;
+            }
+            ++ [
+              self.darwinModules.flakeOverrides
+
+              ({ lib, ... }: {
+                nixpkgs.source = lib.mkDefault nixpkgs;
+
+                system.darwinVersionSuffix = ".${self.shortRev or "dirty"}";
+                system.darwinRevision = lib.mkIf (self ? rev) self.rev;
+              })
```

```suggestion
              ({ lib, ... }: {
                nixpkgs.source = lib.mkDefault nixpkgs;
                
                system.checks.verifyNixPath = lib.mkDefault false;
                system.checks.verifyNixChannels = lib.mkDefault false;

                system.darwinVersionSuffix = ".${self.shortRev or "dirty"}";
                system.darwinRevision = lib.mkIf (self ? rev) self.rev;
              })
```

Might be worth just dropping the `flake-overrides` module altogether

---

Sure.